mvpoly/src/prime.rs

```diff
@@ -585,4 +592,138 @@ impl<F: PrimeField, const N: usize, const D: usize> From<F> for Dense<F, N, D> {
+                )
+            }
+            ConstantExprInner::Constant(ConstantTerm::Mds {
+                row: _row,
+                col: _col,
+            }) => {
+                unimplemented!("The constant Mds is not supposed to be used in this context")
+            }
+            ConstantExprInner::Constant(ConstantTerm::Literal(c)) => Dense::from(c),
+        }
+    }
+}
+
+impl<F: PrimeField, const N: usize, const D: usize> From<Operations<ConstantExprInner<F>>>
+    for Dense<F, N, D>
+{
+    fn from(op: Operations<ConstantExprInner<F>>) -> Self {
+        use kimchi::circuits::expr::Operations::*;
+        match op {
+            Atom(op_const) => Self::from(op_const),
+            Add(c1, c2) => Self::from(*c1) + Self::from(*c2),
+            Sub(c1, c2) => Self::from(*c1) - Self::from(*c2),
+            Mul(c1, c2) => Self::from(*c1) * Self::from(*c2),
+            Square(c) => Self::from(*c.clone()) * Self::from(*c),
+            Double(c1) => Self::from(*c1).double(),
+            Pow(c, e) => {
+                // FIXME: dummy implementation
+                let p = Dense::from(*c);
+                let mut result = p.clone();
+                for _ in 0..e {
+                    result = result.clone() * p.clone();
+                }
+                result
+            }
+            Cache(_c, _) => {
+                unimplemented!("The module prime is supposed to be used for generic multivariate expressions, not tied to a specific use case like Kimchi with this constructor")
+            }
+            IfFeature(_c, _t, _f) => {
+                unimplemented!("The module prime is supposed to be used for generic multivariate expressions, not tied to a specific use case like Kimchi with this constructor")
+            }
+        }
+    }
+}
+
+impl<Column: Into<usize>, F: PrimeField, const N: usize, const D: usize>
+    From<Expr<ConstantExpr<F>, Column>> for Dense<F, N, D>
+{
+    fn from(expr: Expr<ConstantExpr<F>, Column>) -> Self {
+        // This is a dummy implementation
+        // TODO: Implement the actual conversion logic
+        use kimchi::circuits::expr::Operations::*;
+
+        match expr {
+            Atom(op_const) => {
+                match op_const {
+                    ExprInner::UnnormalizedLagrangeBasis(_) => {
+                        unimplemented!("Not used in this context")
+                    }
+                    ExprInner::VanishesOnZeroKnowledgeAndPreviousRows => {
+                        unimplemented!("Not used in this context")
+                    }
+                    ExprInner::Constant(c) => Self::from(c),
+                    ExprInner::Cell(Variable { col, row }) => {
+                        assert_eq!(row, CurrOrNext::Curr, "Only current row is supported for now. You cannot reference the next row");
+                        Self::from_variable(col)
+                    }
+                }
+            }
+            Add(e1, e2) => {
```

Comment:
there is no way to not duplicate eg. this add, as it is already there for inner expression ?